### ISSUE TITLE
Visual Studio Code DB2 Connect [Mac] Displays "command 'extension.Db2ManageConnProfiles' not found" when trying to manage connection profiles

### ISSUE THREAD

**Issue by rferin** [ibmdb/vscode-extension#22]
Hello,

I am running Visual Studio Code on a Mac.  I am trying to install "DB2Connect [mac version].  I can install the plug in but it does not work.

When I select "DB2: Manage Connection Profiles", this error is displayed:
command 'extension.Db2ManageConnProfiles' not found

When I select "DB2: Rebuild Native Modules, this error is displayed:
command 'extension.Db2RebuildNativeModules' not found

I have tried uninstalling and reinstalling the connector, but the same error persists.  I have tried rebooting my machine, but the same error persists.  I have tried searching for a solution but I have not found one that works for me.

Here is the information for my VSC environment:

Version: 1.42.1
Date: 2020-02-11T14:44:27.652Z
Electron: 6.1.6
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.6.0

Please let me know how I can correct the issue.
Thank you.

**Comment by akhilravuri1** [comment]
Hi @rferin 

Please give us some information like..
node and npm are installed?

If installed please  test us what is the version of node and npm

run node -v and npm -v in command prompt. (for vscode 1.42 please install node 12.4.0 ).

Please give us the log even after installing the node if you are facing the issue.

How to take a log?
see this comment..https://github.com/ibmdb/vscode-extension/issues/20#issuecomment-694261954

Thanks,
Akhil

**Comment by rferin** [comment]
Hello Akhil,
These are the versions I have.  I will work on getting the log.

% node -v
v12.16.0
% npm -v
6.13.4
% git --version
git version 2.28.0

Thanks

**Comment by rferin** [comment]
Hello Akhil,

Here is the output from the terminal:

[9/29/2020 11:17:41 AM] Proceeding to rebuild native modules
[9/29/2020 11:17:41 AM] No ibm_db found...
Running npm install,ibm_db in /Users/me/.vscode/extensions/ibm.db2connect-1.1.0...

> ibm_db@2.7.1 install /Users/me/.vscode/extensions/ibm.db2connect-1.1.0/node_modules/ibm_db
> node installer/driverInstall.js
platform =  darwin , arch =  x64 , node.js version =  v12.16.0
make version = GNU Make 3.81

Proceeding to build IBM_DB for Electron framework...
Unable to detect electon installation.
/bin/sh: code: command not found
Unable to find VSCode version, will use Electron version  9.2.1
Downloading DB2 ODBC CLI Driver from https://public.dhe.ibm.com/ibmdl/export/pub/software/data/db2/drivers/odbc_cli/macos64_odbc_cli.tar.gz...

0.03% | 8192 bytes downloaded out of 28394652 bytes.

0.06% | 16384 bytes downloaded out of 28394652 bytes.
100.00% | 28394652 bytes downloaded out of 28394652 bytes.

****************************************
You are downloading a package which includes the Node.js module for IBM DB2/Informix.  The module is licensed under the Apache License 2.0. The package also includes IBM ODBC and CLI Driver from IBM, which is automatically downloaded as the node module is installed on your system/device. The license agreement to the IBM ODBC and CLI Driver is available in undefined   Check for additional dependencies, which may come with their own license agreement(s). Your use of the components of the package and dependencies constitutes your acceptance of their respective license agreements. If you do not accept the terms of any license agreement(s), then delete the relevant component(s) from your device.
****************************************
Downloading and extraction of DB2 ODBC CLI Driver completed successfully ...

Error: Command failed: node-gyp configure build  --IS_DOWNLOADED=true --target=9.2.1 --arch=x64 --dist-url=https://atom.io/download/electron --IBM_DB_HOME="$IBM_DB_HOME"
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.
No receipt for 'com.apple.pkg.DeveloperToolsCLILeo' found at '/'.
No receipt for 'com.apple.pkg.DeveloperToolsCLI' found at '/'.
gyp: No Xcode or CLT version detected!
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:321:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build" "--IS_DOWNLOADED=true" "--target=9.2.1" "--arch=x64" "--dist-url=https://atom.io/download/electron" "--IBM_DB_HOME=/Users/me/.vscode/extensions/ibm.db2connect-1.1.0/node_modules/ibm_db/installer/clidriver"
gyp ERR! cwd /Users/me/.vscode/extensions/ibm.db2connect-1.1.0/node_modules/ibm_db
gyp ERR! node -v v12.16.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
    at ChildProcess.exithandler (child_process.js:303:12)
    at ChildProcess.emit (events.js:321:20)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5) {
  killed: false,
  code: 1,
  signal: null,
  cmd: 'node-gyp configure build  --IS_DOWNLOADED=true --target=9.2.1 --arch=x64 --dist-url=https://atom.io/download/electron --IBM_DB_HOME="$IBM_DB_HOME"'
}
npm 
WARN Db2connect@1.1.0 No repository field.
npm WARN Db2connect@1.1.0 No license field.

npm 
ERR! code ELIFECYCLE
npm ERR! 
errno 1
npm ERR! ibm_db@2.7.1 install: `node installer/driverInstall.js`
npm ERR!
 Exit status 1
npm ERR!
 
npm ERR!
 Failed at the ibm_db@2.7.1 install script.
npm ERR!
 This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2020-09-29T15_17_59_908Z-debug.log
[9/29/2020 11:17:59 AM] ibm_db installation failed with code: 1
Error: Failed to execute command "npm" with  arguments "install, ibm_db" return code 1

**Comment by akhilravuri1** [comment]
Hi @rferin 

can you run code --version in your command prompt?

If code is not available from the terminal, please add it to the env so that it will be available from terminal.

thanks,
Akhil

**Comment by akhilravuri1** [comment]
And please install node 12.4.0 as recommended by vscode

**Comment by rferin** [comment]
Hi Akhil,

Code is not available:
zsh: command not found: code

What is code?  I searched in finder but could not find it.

**Comment by akhilravuri1** [comment]
code is the vscode command.

**Comment by akhilravuri1** [comment]
Please take a look at this https://stackoverflow.com/questions/29955500/code-not-working-in-command-line-for-visual-studio-code-on-osx-mac

**Comment by rferin** [comment]
Hi Akhil,

I added code to the path, now I get this:

code --version
1.42.1
c47d83b293181d9be64f27ff093689e8e7aed054
x64

I also reinstalled Xcode.

The DB2 connector still fails with the same error.

The about for visual studio code lists node 12.4.0.
Node.js: 12.4.0

Is this OK?
I did look on nodejs.org, but I am not sure which is the correct package for Mac.

**Comment by akhilravuri1** [comment]
Hi @rferin 

Is it possible to update the vscode?

Because it is too old and we have removed the binary.
Sorry, I should have told you before...but I just cross-checked now and found that.

Thanks,
Akhil

**Comment by akhilravuri1** [comment]
If you are planning to download the lasted vscode..1.49 or 1.49.x

Please download the node.js from here https://nodejs.org/download/release/v12.14.1/
You can download .pkg or tar file.

After installing the node.js if the vscode is opened... please close it and open it again.
Now install the extension.

If you are not planning to update the vscode please let me know.

Thanks,

**Comment by rferin** [comment]
Hello Akhil,

This morning I uninstalled and reinstalled Xcode, but that didn't help.

Then I updated to the latest Visual Studio code:

Version: 1.49.2
Commit: e5e9e69aed6e1984f7499b7af85b3d05f9a6883a
Date: 2020-09-24T16:23:52.277Z
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0

Here are the versions of node and npm.
 node -v
v12.14.1
 % npm -v
6.13.4

It still did not work so I re-installed Xcode again.  Now it appears to be working.  I haven't tried to connect to DB2 yet, but finally the "Connection Profile" panel is appearing.
